### PR TITLE
fixed all the user bugs

### PR DESCRIPTION
**Pull request by AnikaAgiwal2711**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 80% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: powerflex/provider/user_resource.go [hunk 1]
```diff
@@ -287,3 +302,17 @@ func (r *userResource) Delete(ctx context.Context, req resource.DeleteRequest, r
...
+
+func (r *userResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+	parts := strings.Split(req.ID, ":")
+	var userName string
+    if len(parts) == 2 {
+        userName = parts[1]
+    } else {
+		resp.Diagnostics.AddError(
+            "Unexpected Import Identifier",
+            fmt.Sprintf("Expected import identifier with format: name:username. Got: %q", req.ID),
```
Comment (rounak-adhikary) [inline powerflex/provider/user_resource.go:314]: I am a bit confused by this import format.
If we allowed the user to import by both id and name, then we cd have asked the user to give the id as "id:<id>" and name as "name:<name>".
But here, we are only allowing the user to give names to us, no id. So why force the user to type the "name:" prefix? An import format of "<name>" seems sufficient.

Another observation in this regard is that, we are not checking the prefix at all, which means that if the user gave "whocares:<name>" or "id:<name>" as the import id, the import wd still work.

Comment (shenda1) [reply powerflex/provider/user_resource.go:314]: @rounak-adhikary Are we supporting import by multiple attributes in any resource? If yes, we have to be consistent in the format.

Comment (rounak-adhikary) [reply powerflex/provider/user_resource.go:314]: Import is supported in ome devices resource by both id and service tags. Ref: https://github.com/dell/terraform-provider-ome/blob/main/docs/resources/devices.md#import
Imported is supported in many ome resources by only name, example: [template](https://github.com/dell/terraform-provider-ome/blob/main/docs/resources/template.md#import) and [static group](https://github.com/dell/terraform-provider-ome/blob/main/docs/resources/static_group.md#import)

PS: My point is not that we should support import by multiple indices, but rather that, in cases where import by only one particular index is supported, we do not use the "<index_type>:<index_value>" import ID format. We just use "<index_value>" format.

Comment (AnikaAgiwal2711) [reply powerflex/provider/user_resource.go:314]: Another observation in this regard is that, we are not checking the prefix at all, which means that if the user gave "whocares:[name]" or "id:[name]" as the import id, the import wd still work.

This is a good point.

Comment (AnikaAgiwal2711) [reply powerflex/provider/user_resource.go]: @rounak-adhikary : we have decided to provide support for both id and name and this is the below format:
terraform import powerflex_user.block1 "id:id1"
terraform import powerflex_user.block1 "name:name1"

Let me know if that looks good to you or not!!

Comment (rounak-adhikary) [reply powerflex/provider/user_resource.go]: That sounds good!

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Review by rounak-adhikary — APPROVED** [review]
LGTM

**Comment by Krunal-Thakkar** [comment]
run e2e test

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Comment by AnikaAgiwal2711** [comment]
run e2e test

**Review by Krunal-Thakkar — APPROVED** [review]

**Pull request merged**